cosmpy/aerial/contract/__init__.py

```diff
@@ -182,38 +184,45 @@ def store(
 
     def instantiate(
         self,
-        code_id: int,
         args: Any,
         sender: Wallet,
         label: Optional[str] = None,
         gas_limit: Optional[int] = None,
         admin_address: Optional[Address] = None,
         funds: Optional[str] = None,
     ) -> Address:
-        """instantiate the contract.
+        """Instantiate the contract.
 
-        :param code_id: code id
         :param args: args
         :param sender: sender wallet address
         :param label: label, defaults to None
         :param gas_limit: transaction gas limit, defaults to None
         :param admin_address: admin address, defaults to None
         :param funds: funds, defaults to None
         :raises RuntimeError: Unable to extract contract code id
+
         :return: contract address
         """
-        assert self._digest is not None
+        assert self._code_id, RuntimeError("Code id was not set.")
 
         if self._schema is not None:
             validate(args, self._schema[INSTANTIATE_MSG])
 
-        label = label or _generate_label(bytes(self._digest))
+        if label is None:
+            if self._digest:
+                label = _generate_label(bytes(self._digest))
+            elif self._code_id:
+                label = f"{self._code_id}--{datetime.utcnow().strftime('%Y%m%d%H%M%S')}"
```

Comment:
Very minor point, but it seems like it would be more consistent to either have both of these cases call functions (`_generate_label_by_...`) or neither.